### PR TITLE
Fix interaction with drag and infowindows

### PR DESCRIPTION
**Pull request by Xatpy**
The problem was that it doesn’t ‘wait’ for the drag stops to launch the infowindow.
It works for Android and iOS (touch start)

**Incorrect**
interactivity_incorrect

--> This fix solves it -->
**Correct**
interactivity_correct

## Tested with:
- Devices: OnePlusOne (Android 4.4/Chrome) and iPhone 4S (iOS 7 / Safari-Chrome).
- [Viz.json](http://team.cartodb.com/api/v2/viz/c4d2e894-b685-11e4-994e-0e0c41326911/viz.json)
- Leaflet / GMaps

# Why it isn't working with Internet Explorer 11 and Windows Mobile Phone 8.1
*It can't detect* **up events** *correctly.* I've done this [example test](http://bl.ocks.org/Xatpy/raw/814d5df9e1911fd83e64/) and *up events* aren't working correctly (it triggers automatically). 

I don't know if it's an IE bug, or a phone bug. I'm using Nokia Lumia 635. [There's another guy with the same problem](https://social.technet.microsoft.com/Forums/ie/en-US/0e939601-c14c-4637-b0c8-08277e850900/ie11-mobile-windows-phone-81-touchend-fired-without-user-action?forum=ieitprocurrentver).

There's another issue: the number of pointerdown events launched depends on viz.json; and I don't know why. So, I can't manage it. I've done several tests (with a console.log in function onDown() and onUp() printing the type) and with a viz.json with several layers. Depending on the number of layers (**and its visibility**), those events are triggered. I could do "hacks" that allow to work GMaps and Leaflet separately, but no together, so it isn't commited. Also, if Google Maps is being used the mousedown event is triggered (extra).

### CODE REVIEW

**Comment by Xatpy** [comment]
Please, @javisantana , could you check it? Thanks!

File: control/lib/interaction.js [hunk 1]
```diff
@@ -117,10 +117,12 @@ wax.interaction = function() {
...
-            // Don't make the user click close if they hit another tooltip
-            bean.fire(interaction, 'off');
-            // Touch moves invalidate touches
-            bean.add(parent(), touchEnds);
+            //GMaps fix: Because it's triggering always mousedown and click, we've to remove it
+            bean.remove(document.body, 'click', onUp); //GMaps fix
+
+            //When we finish dragging, then the click will be 
+            bean.add(document.body, 'click', onUp);
+            bean.add(document.body, 'touchEnd', dragEnd);
```
Comment (javisantana) [inline control/lib/interaction.js]: why document.body instead of parent() ?

Comment (Xatpy) [inline control/lib/interaction.js]: Because of [this](https://github.com/Xatpy/wax/blob/fix_interaction_remove/control/lib/interaction.js#L113)

**Comment by Xatpy** [comment]
I've opened a [post in StackOverflow](http://stackoverflow.com/questions/28625604/pointerup-event-isnnt-working-correctly-with-winmobile8-1-ie11) looking for a workaround about pointerup and IE11.

Meanwhile, do we merge this PR?

**Comment by javisantana** [comment]
yes, when you merge in cartodb.js please, make test in all desktop browsers in a visuailzation with multple layers

**Comment by Xatpy** [comment]
Great! I'm testing, and if everything is ok I'll merge and release on Monday.

**Comment by Xatpy** [comment]
Closing with [commit](https://github.com/CartoDB/wax/commit/66da964cd4eae6e753226cf48270b15899761a27)

**Pull request closed without merging**